quickwit/quickwit-serve/src/index_api/rest_handler.rs

```diff
@@ -141,6 +145,98 @@ fn get_indexes_metadatas_handler(
         .map(make_response)
 }
 
+// This structure represents describe Index
+#[derive(Serialize, Deserialize, utoipa::ToSchema)]
+struct IndexStats {
+    pub index_id: String,
+    pub index_uri: Uri,
+    pub num_published_splits: usize,
+    pub num_published_docs: u64,
+    pub size_published_docs: Byte,
+    pub timestamp_field_name: Option<String>,
+    pub min_timestamp: Option<i64>,
+    pub max_timestamp: Option<i64>,
+}
+
+#[utoipa::path(
+    get,
+    tag = "Indexes",
+    path = "/indexes/{index_id}/describe",
+    responses(
+        (status = 200, description = "Successfully fetched stats about Index.", body = IndexStats)
+    ),
+    params(
+        ("index_id" = String, Path, description = "The index ID to retrieve index metadata and split data."),
+    )
+)]
+
+// Get describe index
```

Comment:
For function docs, we prefer this format:

```suggestion
/// Describes an index.
```